checker/src/org/checkerframework/checker/units/UnitsRelationsTools.java

```diff
@@ -178,17 +207,42 @@ private static boolean hasNoPrefix(/*@Nullable*/ final AnnotationValue annotatio
     }
 
     /**
-     * Removes the Prefix value from an Annotation, by constructing and returning a copy of its base SI unit's Annotation
-     * @param elements
-     *            Element Utilities from a checker's processing environment,
-     *            typically obtained by calling env.getElementUtils() in init()
-     *            of a Units Relations implementation
-     * @param unitsAnnotation An AnnotationMirror representing a Units Annotation
-     * @return The base SI Unit's AnnotationMirror, or null if the base SI Unit cannot
-     *         be constructed
+     * Given an Annotated Type, this method returns the Annotation representing
+     * the unit that this Annotated Type has.
+     *
+     * @param annoType an Annotated Type
+     * @return an AnnotationMirror representing the unit that this Annotated
+     *         Type has
+     */
+    public static /*@Nullable*/ AnnotationMirror getUnit(/*@Nullable*/ final AnnotatedTypeMirror annoType) {
+        if (annoType == null) {
+            return null;
+        }
+        // return the first unit (should always be the only unit) in the
+        // annotated type
+        Iterator<AnnotationMirror> it = annoType.getEffectiveAnnotations().iterator();
+        if (it.hasNext()) {
+            return annoType.getEffectiveAnnotations().iterator().next();
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Removes the Prefix value from an Annotation, by constructing and
+     * returning a copy of its base SI unit's Annotation
+     *
+     * @param elements Element Utilities from a checker's processing
+     *            environment, typically obtained by calling
+     *            env.getElementUtils() in init() of a Units Relations
+     *            implementation
+     * @param unitsAnnotation An AnnotationMirror representing a Units
+     *            Annotation
+     * @return The base SI Unit's AnnotationMirror, or null if the base SI Unit
+     *         cannot be constructed
      */
     public static /*@Nullable*/ AnnotationMirror removePrefix(/*@Nullable*/ final Elements elements, /*@Nullable*/ final AnnotationMirror unitsAnnotation) {
-        if (elements == null) {
+        if (elements == null || unitsAnnotation == null) {
```

Comment:
Why allow `null`? Why not raise an error and debug the defaulting rules and users of the method?